uberenv.py

```diff
@@ -151,6 +151,13 @@ def parse_args():
                       default=None,
                       help="override the default package name")
 
+    # uberenv tpl build mode
+    parser.add_option("--build-mode",
+                      dest="build_mode",
+                      default=None,
+                      help="set mode used to build third party dependencies "
+                           "(spack options: 'dev-build' 'uberenv-pkg')")
```

Comment:
@cyrush -- a comment that you posted indicates that `install` is also an option.  Should it be added to this help line as well?

Reply:
This seems like a much cleaner way to expose this functionality!

Reply:
Looking through the code, it appears that `install` will be used if one does not set `--build-mode`,
so, perhaps the `help` should indicate that?

Reply:
It's actually more complex than that :-(

The default `build_mode` should effectively be `dev-build`

The reason why it's complex has to do with the logic for command line args vs json overrides.

There is no one place for defaults, if you add a default in the command line arg def -- it will always override the json key (which isn't what we want).